lightningd/options.c

```diff
@@ -266,6 +279,8 @@ static void dev_register_opts(struct lightningd *ld)
 			 "Time between gossip broadcasts in milliseconds (default: 30000)");
 	opt_register_arg("--dev-disconnect=<filename>", opt_subd_dev_disconnect,
 			 NULL, ld, "File containing disconnection points");
+	opt_register_arg("--dev-hsm-seed=<seed>", opt_set_hsm_seed,
```

Comment:
Seems ok, though it would have been simpler to just have the testing framework write an `hsm_secret` file before starting the daemon.

Reply:
Except we trigger other things on whether the directory exists or not, which seems less error-prone.